### PR TITLE
Bump to most recent ruby/jruby; drop ruby no longer supported

### PR DESCRIPTION
**Pull request by joshRpowell**
a little housekeeping in an effort to get the ball rolling again.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -7,14 +7,14 @@ os:
   - osx
 
 rvm:
-  - "2.0.0"
-  - "2.1" # latest 2.1.x
-  - "2.2.5"
-  - "2.3.3"
-  - "2.4.0"
+  - "2.2.10"
```
Comment (kbrock) [inline .travis.yml]: This is EOL - do we need to support this?  [[ref]](https://www.ruby-lang.org/en/downloads/branches/)

File: .travis.yml [hunk 2]
```diff
@@ -7,14 +7,14 @@ os:
...
+  - "2.3.8"
+  - "2.4.5"
+  - "2.5.3"
   - "ruby-head"
   - "jruby-9.0.5.0"
```
Comment (kbrock) [inline .travis.yml]: defunct

File: .travis.yml [hunk 3]
```diff
@@ -7,14 +7,14 @@ os:
...
-  - "jruby-9.1.5.0"
+  - "jruby-9.1.17.0"
```
Comment (kbrock) [inline .travis.yml]: support ended 2017 [ref](https://github.com/jruby/jruby/wiki/Roadmap)

File: .travis.yml [hunk 4]
```diff
@@ -7,14 +7,14 @@ os:
   - osx
```
Comment (kbrock) [inline .travis.yml]: a question for the maintainers - but osx seem overkill here

Comment (joshRpowell) [reply .travis.yml]: great feedback! committed changes. lets see what the maintainers say about 
```suggestion
  - osx
```

**Comment by ab** [comment]
Thanks for the PR! We tend to take a very conservative approach to dropping support for older versions of ruby, since older versions of ruby tend to be used fairly widely for long after the official EOL dates.

We test on OS X because Apple monkey patches OpenSSL to have different behavior than on other platforms, and rest-client prints some different warnings as a result.

The Travis tests should be working again thanks to #716.

**Pull request closed without merging**